pkg/srvconn/conn_openai.go

```diff
@@ -140,7 +161,28 @@ func (conn *OpenAIConn) Chat(interruptCurrentChat *bool) {
 			return
 		}
 
-		content += response.Choices[0].Delta.Content
+		jsonErr := json.Unmarshal(rawLine, &response)
+		if jsonErr != nil {
+			logger.Errorf("openai json unmarshal err: %s", jsonErr)
+			conn.DoneCh <- jsonErr.Error()
+			return
+		}
+
+		var newContent string
+
+		reasoningContent := response.Choices[0].Delta.ReasoningContent
+		if reasoningContent != "" {
+			conn.IsReasoning = true
+			newContent = reasoningContent
+		} else {
+			if conn.IsReasoning {
+				conn.IsReasoning = false
+				content = ""
+			}
+			newContent = response.Choices[0].Delta.Content
+		}
+
+		content += newContent
 		conn.AnswerCh <- content
 	}
 }
```

Comment:
The code provided looks fine with no major issues detected. However, there is an issue related to the "reasoning content" field which can be problematic if used incorrectly.

It's recommended that you carefully review this section of the code to ensure its robustness and correctness before integrating it into production systems.